<!doctype html>
<!--
  Minimal Mistakes Jekyll Theme 4.22.0 by Michael Rose
  Copyright 2013-2020 Michael Rose - mademistakes.com | @mmistakes
  Free for personal and commercial use under the MIT license
  https://github.com/mmistakes/minimal-mistakes/blob/master/LICENSE
-->
<html lang="en" class="no-js">
  <head>
    <meta charset="utf-8">

<!-- begin _includes/seo.html --><title>Software Tutorial: Finding Local Differences in Two Protein Structures - Xu lab@CBD_SCS_CMU</title>
<meta name="description" content="Write an awesome description for your new site here. You can edit this line in _config.yml. It will appear in your document head meta (for Google search results) and in your feed.xml site description.">



<meta property="og:type" content="website">
<meta property="og:locale" content="en_US">
<meta property="og:site_name" content="Xu lab@CBD_SCS_CMU">
<meta property="og:title" content="Software Tutorial: Finding Local Differences in Two Protein Structures">
<meta property="og:url" content="http://localhost:4000/coronavirus/tutorial_multiseq">


  <meta property="og:description" content="Write an awesome description for your new site here. You can edit this line in _config.yml. It will appear in your document head meta (for Google search results) and in your feed.xml site description.">












<link rel="canonical" href="http://localhost:4000/coronavirus/tutorial_multiseq">




<script type="application/ld+json">
  {
    "@context": "https://schema.org",
    
      "@type": "Person",
      "name": null,
      "url": "http://localhost:4000/"
    
  }
</script>






<!-- end _includes/seo.html -->



<!-- https://t.co/dKP3o1e -->
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<script>
  document.documentElement.className = document.documentElement.className.replace(/\bno-js\b/g, '') + ' js ';
</script>

<!-- For all browsers -->
<link rel="stylesheet" href="/assets/css/main.css">
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@5/css/all.min.css">

<!--[if IE]>
  <style>
    /* old IE unsupported flexbox fixes */
    .greedy-nav .site-title {
      padding-right: 3em;
    }
    .greedy-nav button {
      position: absolute;
      top: 0;
      right: 0;
      height: 100%;
    }
  </style>
<![endif]-->


    <!-- start custom head snippets -->

<!-- insert favicons. use https://realfavicongenerator.net/ -->

<!-- end custom head snippets -->

  </head>

  <body class="layout--single">
    <nav class="skip-links">
  <h2 class="screen-reader-text">Skip links</h2>
  <ul>
    <li><a href="#site-nav" class="screen-reader-shortcut">Skip to primary navigation</a></li>
    <li><a href="#main" class="screen-reader-shortcut">Skip to content</a></li>
    <li><a href="#footer" class="screen-reader-shortcut">Skip to footer</a></li>
  </ul>
</nav>

    <!--[if lt IE 9]>
<div class="notice--danger align-center" style="margin: 0;">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience.</div>
<![endif]-->

    <div class="masthead">
  <div class="masthead__inner-wrap">
    <div class="masthead__menu">
      <nav id="site-nav" class="greedy-nav">
        <a class="site-title" href="/">Xu lab@CBD_SCS_CMU</a>
        <ul class="visible-links">
                
                
                    <li class="masthead__menu-item">
                        <a href="/news/">NEWS</a>
                    </li>
                
            
                
                
                    <li class="masthead__menu-item">
                        <a href="/papers/">PAPERS</a>
                    </li>
                
            
                
                
                    <li class="masthead__menu-item">
                        <a href="/group/">GROUP</a>
                    </li>
                
            
                
                
                    <li class="masthead__menu-item">
                        <a href="/meet-the-team/">ABOUT MIN XU</a>
                    </li>
                
            
                
                
                    <li class="masthead__menu-item">
                        <a href="/position/">POSITION</a>
                    </li>
                
            
                
                
                    <li class="masthead__menu-item">
                        <a href="/software/">SOFTWARE</a>
                    </li>
                
            
                
                
                    <li class="dropdown ">
                        <a class="dropdown-toggle" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false">Take the Course    <i class="fa fa-caret-down fa-sm" aria-hidden="true"></i><span class="caret"></span></a>
                        <ul class="dropdown-content">
                            
                                <li>
                                    <a href="/prologue/">    <b>Prologue:</b> Introducing biological modeling with Turing patterns</a>
                                </li>
                            
                                <li>
                                    <a href="/motifs/home">    <b>Module 1:</b> Finding motifs in transcription factor networks</a>
                                </li>
                            
                                <li>
                                    <a href="/chemotaxis/home">    <b>Module 2:</b> Unpacking E. coli's genius exploration algorithm</a>
                                </li>
                            
                                <li>
                                    <a href="/coronavirus/home">    <b>Module 3:</b> Analyzing the coronavirus spike protein</a>
                                </li>
                            
                                <li>
                                    <a href="">    <b>Module 4:</b> coming soon!</a>
                                </li>
                            
                        </ul>
                    </li>
                
            
        </ul>
        <!--
        
        <button class="search__toggle" type="button">
          <svg class="icon" width="16" height="16" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 15.99 16">
            <path d="M15.5,13.12L13.19,10.8a1.69,1.69,0,0,0-1.28-.55l-0.06-.06A6.5,6.5,0,0,0,5.77,0,6.5,6.5,0,0,0,2.46,11.59a6.47,6.47,0,0,0,7.74.26l0.05,0.05a1.65,1.65,0,0,0,.5,1.24l2.38,2.38A1.68,1.68,0,0,0,15.5,13.12ZM6.4,2A4.41,4.41,0,1,1,2,6.4,4.43,4.43,0,0,1,6.4,2Z" transform="translate(-.01)"></path>
          </svg>
        </button>
        
        -->
        <button class="greedy-nav__toggle hidden" type="button">
          <span class="visually-hidden">Toggle Menu</span>
          <div class="navicon"></div>
        </button>
        <ul class="hidden-links hidden"></ul>
      </nav>
    </div>
  </div>
</div>


    <div class="initial-content">
      



<div id="main" role="main">

  <script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML" type="text/javascript"></script>

  
  <div class="sidebar sticky">
  
  
    
      
      
      
      
    
    
      

<nav class="nav__list">
  
  <input id="ac-toc" name="accordion-toc" type="checkbox" />
  <label for="ac-toc">Toggle Menu</label>
  <ul class="nav__items">
    
      <li>
        
          <span class="nav__sub-title">Introduction</span>
        

        
        <ul>
          
            <li><a href="/coronavirus/home">A Tale of Two Doctors<br><br></a></li>
          
        </ul>
        
      </li>
    
      <li>
        
          <span class="nav__sub-title">Part 1: Protein Structure Prediction</span>
        

        
        <ul>
          
            <li><a href="/coronavirus/structure_intro">An Introduction to Protein Structure Prediction<br><br></a></li>
          
            <li><a href="/coronavirus/ab_initio">Ab initio Protein Structure Prediction<br><br></a></li>
          
            <li><a href="/coronavirus/homology">Homology Modeling for Protein Structure Prediction<br><br></a></li>
          
            <li><a href="/coronavirus/accuracy">Comparing Protein Structures to Assess Model Accuracy<br><br></a></li>
          
            <li><a href="/coronavirus/conclusion_part_1">Part 1 Conclusion: Protein Structure Prediction is Solved! (Kinda)<br><br></a></li>
          
        </ul>
        
      </li>
    
      <li>
        
          <span class="nav__sub-title">Part 2: Comparing SARS-CoV-2 and SARS</span>
        

        
        <ul>
          
            <li><a href="/coronavirus/multiseq">Searching for Local Differences in the SARS-CoV and SARS-CoV-2 Spike Proteins<br><br></a></li>
          
            <li><a href="/coronavirus/structural_diff">Analyzing Structural Differences in the Bonding of SARS-CoV and SARS-CoV-2 with the ACE2 Enzyme<br><br></a></li>
          
            <li><a href="/coronavirus/NAMD">Quantifying the Interaction Energy Between the SARS-CoV-2 Spike Protein and ACE2<br><br></a></li>
          
            <li><a href="/coronavirus/conclusion_part_2">Part 2 Conclusion: From Static Protein Analysis to Molecular Dynamics<br><br></a></li>
          
        </ul>
        
      </li>
    
      <li>
        
          <span class="nav__sub-title">Software Tutorials (featuring ProDy)</span>
        

        
        <ul>
          
            <li><a href="/coronavirus/tutorial_ab_initio">Using ab initio Modeling to Predict the Structure of Hemoglobin Subunit Alpha<br><br></a></li>
          
            <li><a href="/coronavirus/tutorial_homology">Using Homology Modeling to Predict the Structure of the SARS-CoV-2 Spike Protein<br><br></a></li>
          
            <li><a href="/coronavirus/tutorial_rmsd">Using RMSD to Compare Two Protein Structures<br><br></a></li>
          
            <li><a href="/coronavirus/tutorial_multiseq" class="active">Finding Local Differences in Two Protein Structures<br><br></a></li>
          
            <li><a href="/coronavirus/tutorial_visualization">Visualizing Specific Regions of Interest within a Protein Structure<br><br></a></li>
          
            <li><a href="/coronavirus/tutorial_NAMD">Computing the Energy Contributed by a Local Region of a Bound Complex<br><br></a></li>
          
            <li><a href="/coronavirus/tutorial_GNM">Molecular Dynamics Analysis with GNM<br><br></a></li>
          
            <li><a href="/coronavirus/tutorial_ANM">ANM Simulations Add Directionality to GNM<br><br></a></li>
          
        </ul>
        
      </li>
    
  </ul>
</nav>

    
  
  </div>



  <article class="page" itemscope itemtype="https://schema.org/CreativeWork">
    <meta itemprop="headline" content="Software Tutorial: Finding Local Differences in Two Protein Structures">
    
    
    

    <div class="page__inner-wrap">
      
        <header>
          <h1 id="page-title" class="page__title" itemprop="headline">Software Tutorial: Finding Local Differences in Two Protein Structures
</h1>
          
        </header>
      

      <section class="page__content" itemprop="text">
        
          <aside class="sidebar__right sticky">
            <nav class="toc">
              <header><h4 class="nav__title"><i class="fas fa-file-alt"></i> On this page</h4></header>
              <ul class="toc__menu"><li><a href="#getting-started">Getting started</a></li><li><a href="#aligning-the-rbd-regions-of-two-spike-proteins">Aligning the RBD regions of two spike proteins</a></li><li><a href="#visualizing-a-structural-alignment">Visualizing a structural alignment</a></li></ul>

            </nav>
          </aside>
        
        <p>In this tutorial, we will get started with VMD and then calculate Qres between the SARS-CoV-2 RBD (PDB entry: <a href="https://www.rcsb.org/structure/6vw1" target="_blank">6vw1</a>) and SARS-CoV RBD (PDB entry: <a href="https://www.rcsb.org/structure/2ajf" target="_blank">2ajf</a>) using the VMD plugin Multiseq. By locating regions with low Qres, we can hopefully identify regions of structural differences between the two RBDs.</p>

<p>Multiseq aligns two protein structures using a tool called <strong>Structural Alignment of Multiple Proteins (STAMP)</strong>. Much like the Kabsch algorithm considered in part 1 of the module, STAMP minimizes the distance between alpha carbons of the aligned residues for each protein or molecule by applying rotations and translations. If the structures do not have common structures, then STAMP will fail. For more details on the algorithm used by STAMP, click <a href="http://www.compbio.dundee.ac.uk/manuals/stamp.4.4/stamp.pdf" target="_blank">here</a>.</p>

<h2 id="getting-started">Getting started</h2>

<p>For this tutorial, first <a href="https://www.ks.uiuc.edu/Development/Download/download.cgi?PackageName=VMD" target="_blank">download VMD</a>. Throughout this tutorial, the program may prompt you to download additional protein database information, which you should accept.</p>

<p>We will need to download the <code class="language-plaintext highlighter-rouge">.pdb</code> files for 6vw1 and 2ajf. Visit the <a href="https://www.rcsb.org/structure/6vw1" target="_blank">6vw1</a> and <a href="https://www.rcsb.org/structure/2ajf" target="_blank">2ajf</a> PDB pages. For each protein,  click on <code class="language-plaintext highlighter-rouge">Download Files</code> and select <code class="language-plaintext highlighter-rouge">PDB Format</code>. The following figure shows this for 6vw1.</p>

<p style="font-size: medium;"><img src="../assets/images/Ridge0.png" alt="image-center" class="align-center" /></p>

<h2 id="aligning-the-rbd-regions-of-two-spike-proteins">Aligning the RBD regions of two spike proteins</h2>

<p>Next, launch VMD, which will open three windows. We will not use <code class="language-plaintext highlighter-rouge">VMD.exe</code>, the console window, in this tutorial. We will load molecules and change visualizations in <code class="language-plaintext highlighter-rouge">VMD Main</code>. Finally, we will use <code class="language-plaintext highlighter-rouge">OpenGL Display</code> to display our visualizations.</p>

<p>We will first load the SARS-CoV-2 RBD (6vw1) into VMD. In <code class="language-plaintext highlighter-rouge">VMD Main</code>, go to <code class="language-plaintext highlighter-rouge">File &gt; New Molecule</code>. Click on <code class="language-plaintext highlighter-rouge">Browse</code>, select your downloaded file (<code class="language-plaintext highlighter-rouge">6vw1.pdb</code>) and click <code class="language-plaintext highlighter-rouge">Load</code>.</p>

<p style="font-size: medium;"><img src="../assets/images/Ridge1.png" alt="image-center" class="align-center" /></p>
<p style="font-size: medium;"><img src="../assets/images/Ridge2.png" alt="image-center" class="align-center" /></p>

<p>The molecule should now be listed in <code class="language-plaintext highlighter-rouge">VMD Main</code>, with its visualization in <code class="language-plaintext highlighter-rouge">OpenGL Display</code>.</p>

<p style="font-size: medium;"><img src="../assets/images/Ridge3.png" alt="image-center" class="align-center" /></p>

<p>In the <code class="language-plaintext highlighter-rouge">OpenGL Display</code> window, you can click and drag the molecule to change its orientation. Pressing ‘r’ on your keyboard allows you to rotate the molecule, pressing ‘t’ allows you to translate the molecule, and pressing ‘s’ allows you to enlarge or shrink the molecule (or you can use your mouse’s scroll wheel). Note that left click and right click have different actions.</p>

<p>We now will need to load the SARS-CoV RBD (2ajf). Repeat the above steps for <code class="language-plaintext highlighter-rouge">2ajf.pdb</code>.</p>

<p>After both molecules are loaded into VMD, start up Multiseq by clicking on <code class="language-plaintext highlighter-rouge">Extensions &gt; Analysis &gt; Multiseq</code>.</p>

<p style="font-size: medium;"><img src="../assets/images/Qres1.png" alt="image-center" class="align-center" /></p>

<p>You will see all the chains listed out per file. Both PDB files contain two biological assemblies of the structure. The first is made up of Chain A (ACE2) and Chain E (RBD), and the second is Chain B (ACE2) and Chain F (RBD). Because Chain A is identical to Chain B, and Chain E is identical to Chain F, we only need to work with one assembly. (We will use the second assembly.)</p>

<p style="font-size: medium;"><img src="../assets/images/Qres2.png" alt="image-center" class="align-center" /></p>

<p>Because we only want to compare the RBD, we will only keep chain F of each structure. To remove the other chains, select the chain and click <code class="language-plaintext highlighter-rouge">Edit &gt; Cut</code>.</p>

<p style="font-size: medium;"><img src="../assets/images/Qres3.png" alt="image-center" class="align-center" /></p>

<p>Click <code class="language-plaintext highlighter-rouge">Tools &gt; Stamp Structural Alignment</code>, and a new window will open up.</p>

<p style="font-size: medium;"><img src="../assets/images/Qres4.png" alt="image-center" class="align-center" /></p>

<p>Keep all the values and click <code class="language-plaintext highlighter-rouge">OK</code>; once you have done so, the RBD regions will have been aligned.</p>

<p style="font-size: medium;"><img src="../assets/images/Qres5.png" alt="image-center" class="align-center" /></p>

<h2 id="visualizing-a-structural-alignment">Visualizing a structural alignment</h2>

<p>Now that we have aligned the two RBD regions, we would like to compare their Qres values over the entire RBD. To see a coloring of the protein alignment based on Qres, click <code class="language-plaintext highlighter-rouge">View &gt; Coloring &gt; Qres</code>.</p>

<p style="font-size: medium;"><img src="../assets/images/Qres6.png" alt="image-center" class="align-center" /></p>

<p>Blue indicates a high value of Qres, meaning that the protein structures are similar at this position; red indicates low Qres and dissimilar protein structures.</p>

<p style="font-size: medium;"><img src="../assets/images/Qres7.png" alt="image-center" class="align-center" /></p>

<p>The <code class="language-plaintext highlighter-rouge">OpenGL Display</code> window will now color the superimposed structures according to the value of Qres.</p>

<p style="font-size: medium;"><img src="../assets/images/Qres8.png" alt="image-center" class="align-center" /></p>

<p>We are looking for regions of consecutive amino acids having low Qres, which correspond to locations in which the coronavirus RBDs differ structurally. You may like to explore the alignments yourself to look for regions of interest before we head back to the main text and discuss our results.</p>

<p style="font-size: 100%; text-align: center;"><a href="multiseq#local-comparison-of-spike-proteins-leads-us-to-a-region-of-interest" class="btn btn--primary btn--large">Return to main text</a></p>

        
      </section>

      <footer class="page__meta">
        
        


        

      </footer>

      

      
    </div>

    
  </article>

  <!-- only show related on a post page when `related: true`
  
  
  -->
</div>

    </div>

    
      <div class="search-content">
        <div class="search-content__inner-wrap"><form class="search-content__form" onkeydown="return event.key != 'Enter';">
    <label class="sr-only" for="search">
      Enter your search term...
    </label>
    <input type="search" id="search" class="search-input" tabindex="-1" placeholder="Enter your search term..." />
  </form>
  <div id="results" class="results"></div></div>

      </div>
    

    <div id="footer" class="page__footer">
      <footer>
        <!-- start custom footer snippets -->

<!-- end custom footer snippets -->
        <div class="page__footer-follow">
  <hr>
  <a href="http://cbd.cmu.edu" target="_blank"><img align = "left" src="/biological_modeling_course/assets/images/combined_SCS_Unitmark_187_K.jpg" alt="CBD Logo" width="40%" height="40%"></a>
  <a href="http://mmbios.pitt.edu" target="_blank"><img align="right" src="/biological_modeling_course/assets/images/MMBioS_Logo.png" alt="MMBioS Logo" width="50%" height="50%"></a>
</div>

      </footer>
    </div>

    
  <script src="/assets/js/main.min.js"></script>




<script src="/assets/js/lunr/lunr.min.js"></script>
<script src="/assets/js/lunr/lunr-store.js"></script>
<script src="/assets/js/lunr/lunr-en.js"></script>







  </body>
</html>
