<!doctype html>
<!--
  Minimal Mistakes Jekyll Theme 4.22.0 by Michael Rose
  Copyright 2013-2020 Michael Rose - mademistakes.com | @mmistakes
  Free for personal and commercial use under the MIT license
  https://github.com/mmistakes/minimal-mistakes/blob/master/LICENSE
-->
<html lang="en" class="no-js">
  <head>
    <meta charset="utf-8">

<!-- begin _includes/seo.html --><title>Software Tutorial: Using Homology Modeling to Predict the Structure of the SARS-CoV-2 Spike Protein - Xu lab@CBD_SCS_CMU</title>
<meta name="description" content="Write an awesome description for your new site here. You can edit this line in _config.yml. It will appear in your document head meta (for Google search results) and in your feed.xml site description.">



<meta property="og:type" content="website">
<meta property="og:locale" content="en_US">
<meta property="og:site_name" content="Xu lab@CBD_SCS_CMU">
<meta property="og:title" content="Software Tutorial: Using Homology Modeling to Predict the Structure of the SARS-CoV-2 Spike Protein">
<meta property="og:url" content="http://localhost:4000/coronavirus/tutorial_homology">


  <meta property="og:description" content="Write an awesome description for your new site here. You can edit this line in _config.yml. It will appear in your document head meta (for Google search results) and in your feed.xml site description.">












<link rel="canonical" href="http://localhost:4000/coronavirus/tutorial_homology">




<script type="application/ld+json">
  {
    "@context": "https://schema.org",
    
      "@type": "Person",
      "name": null,
      "url": "http://localhost:4000/"
    
  }
</script>






<!-- end _includes/seo.html -->



<!-- https://t.co/dKP3o1e -->
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<script>
  document.documentElement.className = document.documentElement.className.replace(/\bno-js\b/g, '') + ' js ';
</script>

<!-- For all browsers -->
<link rel="stylesheet" href="/assets/css/main.css">
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@5/css/all.min.css">

<!--[if IE]>
  <style>
    /* old IE unsupported flexbox fixes */
    .greedy-nav .site-title {
      padding-right: 3em;
    }
    .greedy-nav button {
      position: absolute;
      top: 0;
      right: 0;
      height: 100%;
    }
  </style>
<![endif]-->


    <!-- start custom head snippets -->

<!-- insert favicons. use https://realfavicongenerator.net/ -->

<!-- end custom head snippets -->

  </head>

  <body class="layout--single">
    <nav class="skip-links">
  <h2 class="screen-reader-text">Skip links</h2>
  <ul>
    <li><a href="#site-nav" class="screen-reader-shortcut">Skip to primary navigation</a></li>
    <li><a href="#main" class="screen-reader-shortcut">Skip to content</a></li>
    <li><a href="#footer" class="screen-reader-shortcut">Skip to footer</a></li>
  </ul>
</nav>

    <!--[if lt IE 9]>
<div class="notice--danger align-center" style="margin: 0;">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience.</div>
<![endif]-->

    <div class="masthead">
  <div class="masthead__inner-wrap">
    <div class="masthead__menu">
      <nav id="site-nav" class="greedy-nav">
        <a class="site-title" href="/">Xu lab@CBD_SCS_CMU</a>
        <ul class="visible-links">
                
                
                    <li class="masthead__menu-item">
                        <a href="/news/">NEWS</a>
                    </li>
                
            
                
                
                    <li class="masthead__menu-item">
                        <a href="/papers/">PAPERS</a>
                    </li>
                
            
                
                
                    <li class="masthead__menu-item">
                        <a href="/group/">GROUP</a>
                    </li>
                
            
                
                
                    <li class="masthead__menu-item">
                        <a href="/meet-the-team/">ABOUT MIN XU</a>
                    </li>
                
            
                
                
                    <li class="masthead__menu-item">
                        <a href="/position/">POSITION</a>
                    </li>
                
            
                
                
                    <li class="masthead__menu-item">
                        <a href="/software/">SOFTWARE</a>
                    </li>
                
            
                
                
                    <li class="dropdown ">
                        <a class="dropdown-toggle" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false">Take the Course    <i class="fa fa-caret-down fa-sm" aria-hidden="true"></i><span class="caret"></span></a>
                        <ul class="dropdown-content">
                            
                                <li>
                                    <a href="/prologue/">    <b>Prologue:</b> Introducing biological modeling with Turing patterns</a>
                                </li>
                            
                                <li>
                                    <a href="/motifs/home">    <b>Module 1:</b> Finding motifs in transcription factor networks</a>
                                </li>
                            
                                <li>
                                    <a href="/chemotaxis/home">    <b>Module 2:</b> Unpacking E. coli's genius exploration algorithm</a>
                                </li>
                            
                                <li>
                                    <a href="/coronavirus/home">    <b>Module 3:</b> Analyzing the coronavirus spike protein</a>
                                </li>
                            
                                <li>
                                    <a href="">    <b>Module 4:</b> coming soon!</a>
                                </li>
                            
                        </ul>
                    </li>
                
            
        </ul>
        <!--
        
        <button class="search__toggle" type="button">
          <svg class="icon" width="16" height="16" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 15.99 16">
            <path d="M15.5,13.12L13.19,10.8a1.69,1.69,0,0,0-1.28-.55l-0.06-.06A6.5,6.5,0,0,0,5.77,0,6.5,6.5,0,0,0,2.46,11.59a6.47,6.47,0,0,0,7.74.26l0.05,0.05a1.65,1.65,0,0,0,.5,1.24l2.38,2.38A1.68,1.68,0,0,0,15.5,13.12ZM6.4,2A4.41,4.41,0,1,1,2,6.4,4.43,4.43,0,0,1,6.4,2Z" transform="translate(-.01)"></path>
          </svg>
        </button>
        
        -->
        <button class="greedy-nav__toggle hidden" type="button">
          <span class="visually-hidden">Toggle Menu</span>
          <div class="navicon"></div>
        </button>
        <ul class="hidden-links hidden"></ul>
      </nav>
    </div>
  </div>
</div>


    <div class="initial-content">
      



<div id="main" role="main">

  <script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML" type="text/javascript"></script>

  
  <div class="sidebar sticky">
  
  
    
      
      
      
      
    
    
      

<nav class="nav__list">
  
  <input id="ac-toc" name="accordion-toc" type="checkbox" />
  <label for="ac-toc">Toggle Menu</label>
  <ul class="nav__items">
    
      <li>
        
          <span class="nav__sub-title">Introduction</span>
        

        
        <ul>
          
            <li><a href="/coronavirus/home">A Tale of Two Doctors<br><br></a></li>
          
        </ul>
        
      </li>
    
      <li>
        
          <span class="nav__sub-title">Part 1: Protein Structure Prediction</span>
        

        
        <ul>
          
            <li><a href="/coronavirus/structure_intro">An Introduction to Protein Structure Prediction<br><br></a></li>
          
            <li><a href="/coronavirus/ab_initio">Ab initio Protein Structure Prediction<br><br></a></li>
          
            <li><a href="/coronavirus/homology">Homology Modeling for Protein Structure Prediction<br><br></a></li>
          
            <li><a href="/coronavirus/accuracy">Comparing Protein Structures to Assess Model Accuracy<br><br></a></li>
          
            <li><a href="/coronavirus/conclusion_part_1">Part 1 Conclusion: Protein Structure Prediction is Solved! (Kinda)<br><br></a></li>
          
        </ul>
        
      </li>
    
      <li>
        
          <span class="nav__sub-title">Part 2: Comparing SARS-CoV-2 and SARS</span>
        

        
        <ul>
          
            <li><a href="/coronavirus/multiseq">Searching for Local Differences in the SARS-CoV and SARS-CoV-2 Spike Proteins<br><br></a></li>
          
            <li><a href="/coronavirus/structural_diff">Analyzing Structural Differences in the Bonding of SARS-CoV and SARS-CoV-2 with the ACE2 Enzyme<br><br></a></li>
          
            <li><a href="/coronavirus/NAMD">Quantifying the Interaction Energy Between the SARS-CoV-2 Spike Protein and ACE2<br><br></a></li>
          
            <li><a href="/coronavirus/conclusion_part_2">Part 2 Conclusion: From Static Protein Analysis to Molecular Dynamics<br><br></a></li>
          
        </ul>
        
      </li>
    
      <li>
        
          <span class="nav__sub-title">Software Tutorials (featuring ProDy)</span>
        

        
        <ul>
          
            <li><a href="/coronavirus/tutorial_ab_initio">Using ab initio Modeling to Predict the Structure of Hemoglobin Subunit Alpha<br><br></a></li>
          
            <li><a href="/coronavirus/tutorial_homology" class="active">Using Homology Modeling to Predict the Structure of the SARS-CoV-2 Spike Protein<br><br></a></li>
          
            <li><a href="/coronavirus/tutorial_rmsd">Using RMSD to Compare Two Protein Structures<br><br></a></li>
          
            <li><a href="/coronavirus/tutorial_multiseq">Finding Local Differences in Two Protein Structures<br><br></a></li>
          
            <li><a href="/coronavirus/tutorial_visualization">Visualizing Specific Regions of Interest within a Protein Structure<br><br></a></li>
          
            <li><a href="/coronavirus/tutorial_NAMD">Computing the Energy Contributed by a Local Region of a Bound Complex<br><br></a></li>
          
            <li><a href="/coronavirus/tutorial_GNM">Molecular Dynamics Analysis with GNM<br><br></a></li>
          
            <li><a href="/coronavirus/tutorial_ANM">ANM Simulations Add Directionality to GNM<br><br></a></li>
          
        </ul>
        
      </li>
    
  </ul>
</nav>

    
  
  </div>



  <article class="page" itemscope itemtype="https://schema.org/CreativeWork">
    <meta itemprop="headline" content="Software Tutorial: Using Homology Modeling to Predict the Structure of the SARS-CoV-2 Spike Protein">
    
    
    

    <div class="page__inner-wrap">
      
        <header>
          <h1 id="page-title" class="page__title" itemprop="headline">Software Tutorial: Using Homology Modeling to Predict the Structure of the SARS-CoV-2 Spike Protein
</h1>
          
        </header>
      

      <section class="page__content" itemprop="text">
        
          <aside class="sidebar__right sticky">
            <nav class="toc">
              <header><h4 class="nav__title"><i class="fas fa-file-alt"></i> On this page</h4></header>
              <ul class="toc__menu"><li><a href="#swiss-model">SWISS-MODEL</a></li><li><a href="#robetta">Robetta</a></li><li><a href="#galaxyweb">GalaxyWEB</a></li><li><a href="#interpreting-the-results-of-our-software-runs">Interpreting the results of our software runs</a></li></ul>

            </nav>
          </aside>
        
        <p>In this software tutorial, we will apply three popular software resources (SWISS-MODEL, Robetta, and GalaxyWEB) that use homology modeling to predict the structure of the SARS-CoV-2 spike protein. Recall from the main text that this protein is a homotrimer, meaning that it consists of three identical protein structures called chains. In what follows, we will predict the sequence of a single chain.</p>

<p>The details of how the three software resources presented in this lesson <em>differ</em> are beyond the scope of our work in this course. If you are interested in understanding how they each implement homology modeling, then we suggest that you consult the documentation of the appropriate resource.</p>

<h2 id="swiss-model">SWISS-MODEL</h2>
<p>To run SWISS-MODEL, first download the sequence of the spike protein chain: <a href="../_pages/coronavirus/files/CoV2SpikeProteinSeq.txt" download="">SARS-CoV-2 spike protein chain</a>.</p>

<p>Next, go to the main <a href="https://swissmodel.expasy.org/" target="_blank">SWISS-MODEL website</a> and click on <code class="language-plaintext highlighter-rouge">Start Modelling</code>.</p>

<p style="font-size: medium;"><img src="../assets/images/SWISS1.png" alt="image-center" class="align-center" /></p>

<p>On the next page, copy and paste the sequence into the <code class="language-plaintext highlighter-rouge">Target Sequence(s):</code> box. Name your project and enter an email address to get a notification of when your results are ready. Finally, click on <code class="language-plaintext highlighter-rouge">Build Model</code> to submit the job request. Note that you do not need to specify that you want to use the SARS-CoV spike protein as a template because the software will automatically search for a template for you.</p>

<p style="font-size: medium;"><img src="../assets/images/SWISS2.png" alt="image-center" class="align-center" /></p>

<p>Your results may take between an hour and a day to finish depending on how busy the server is. (In the meantime, feel free to run the remaining software.) When you receive an email notification, follow the link provided and you can download the final models.</p>

<p>When we ran our own job, SWISS-MODEL did indeed use one of the PDB entries of SARS-CoV spike protein as its template (PDB: <a href="https://www.rcsb.org/structure/6CRX" target="_blank">6crx</a>) and correctly recognized that the template was a homotrimer. As a result, the software predicted a complete spike protein with all three chains included. An image of our results can be seen below. You can also <a href="../_pages/coronavirus/files/SWISS_Model.zip" download="">download</a> our results. We will discuss how to interpret these results and the <code class="language-plaintext highlighter-rouge">.pdb</code> file format when we return to the main text.</p>

<p style="font-size: medium;"><img src="../assets/images/SWISSResults.png" alt="image-center" class="align-center" />
Structures of the three models of this protein reported by SWISS-MODEL. The superimposed structure of all three models is shown on the bottom right.</p>

<h2 id="robetta">Robetta</h2>
<p>Robetta is a publicly available software resource that uses the same software as the distributed Rosetta@home project that we mentioned earlier in this module. As with SWISS-MODEL, we will provide Robetta a single chain of the SARS-CoV-2 spike protein.</p>

<p>First, if you have not already done so, download the sequence of the chain: <a href="../_pages/coronavirus/files/CoV2SpikeProteinSeq.txt" download="">SARS-CoV-2 spike chain sequence</a>.</p>

<p>Next, visit <a href="https://robetta.bakerlab.org/" target="_blank">Robetta</a> and register for an account.</p>

<p style="font-size: medium;"><img src="../assets/images/Robetta1.png" alt="image-center" class="align-center" /></p>

<p>Then, click on <code class="language-plaintext highlighter-rouge">Structure Prediction &gt; Submit</code>.</p>

<p style="font-size: medium;"><img src="../assets/images/Robetta2.png" alt="image-center" class="align-center" /></p>

<p>Create a name for the job, i.e. “SARS-CoV-2 Spike Chain”. Copy and paste the downloaded sequence into the <code class="language-plaintext highlighter-rouge">Protein sequence</code> box. Check <code class="language-plaintext highlighter-rouge">CM only</code> (for homology modeling), complete the arithmetic problem provided to prove you are human, and then click <code class="language-plaintext highlighter-rouge">Submit</code>.</p>

<p>You should receive an email notification with a link to results after between an hour and a day. In our own run, unlike SWISS-MODEL, Robetta did not deduce that the input protein was a trimer and only predicted a single chain. The structure of the results from our own run of Robetta are shown in the figure below. You can also <a href="../_pages/coronavirus/files/Robetta_Model.zip" download="">download</a> our results if you like.</p>

<p style="font-size: medium;"><img src="../assets/images/RobettaResults.png" alt="image-center" class="align-center" />
The homology models produced by Robetta of one of the chains of the SARS-CoV-2 spike protein. The superimposition of all structures is shown on the bottom right.</p>

<h2 id="galaxyweb">GalaxyWEB</h2>
<p>GalaxyWEB is a server with many available services for protein study, including protein structure prediction. GalaxyTBM (the template-based modeling service) uses <em><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-019-3019-7" target="_blank">HHsearch</a></em> to identify up to 20 templates, and then matches the core sequence with the templates using <em><a href="http://prodata.swmed.edu/promals3d/info/promals3d_help.html" target="_blank">PROMALS3D</a></em>. Next, models are generated using <em><a href="https://pubmed.ncbi.nlm.nih.gov/19089941/" target="_blank">MODELLERCSA</a></em>.</p>

<p>Because GalaxyWEB has a sequence limit of 1000 amino acids, we cannot use the entire spike protein chain. Instead, we will model the receptor binding domain (RBD) of the spike protein, which we introduced in the main text as a variable domain within the spike protein’s S1 subunit.</p>

<p>First, <a href="../_pages/coronavirus/files/CoV2SpikeRBDSeq.txt" download="">download</a> the sequence of the RBD.</p>

<p>Then, visit the <a href="http://galaxy.seoklab.org/" target="_blank">GalaxyWEB</a> homepage. At the top, click on <code class="language-plaintext highlighter-rouge">Services &gt; TBM</code>.</p>

<p style="font-size: medium;"><img src="../assets/images/Galaxy1.png" alt="image-center" class="align-center" /></p>

<p>Enter a job name, i.e. <code class="language-plaintext highlighter-rouge">SARS-CoV-2 RBD</code>. Enter an email address and then copy and paste the RBD sequence into the <code class="language-plaintext highlighter-rouge">SEQUENCE</code> box. Finally, click <code class="language-plaintext highlighter-rouge">Submit</code>.</p>

<p style="font-size: medium;"><img src="../assets/images/Galaxy2.png" alt="image-center" class="align-center" /></p>

<p>You should receive an email notification within a day with a link to your results. The results of our run of GalaxyWEB along with the validated structure of the SARS-CoV-2 RBD (PDB entry: <a href="http://www.rcsb.org/structure/6LZG" target="_blank">6lzg</a>) are visualized in the figure below. You can also <a href="../_pages/coronavirus/files/GalaxyWEB_Models.zip" download="">download</a> our results if you like.</p>

<p style="font-size: medium;"><img src="../assets/images/GalaxyResults.png" alt="image-center" class="align-center" />
Homology models predicted by GalaxyWEB for the SARS-CoV-2 spike protein RBD. The superimposition of all these structures is shown on the bottom right.</p>

<h2 id="interpreting-the-results-of-our-software-runs">Interpreting the results of our software runs</h2>

<p>In the figures above, the structures predicted by the three software resources appear to be reasonably accurate. But throughout this course, we have prioritized using quantitative metrics to analyze results. As we return to the main text, our question is how to develop a quantitative metric to compare the results of these models to each other and to the structure of the SARS-CoV-2 spike protein.</p>

<p style="font-size: 100%; text-align: center;"><a href="homology#applying-homology-modeling-to--the-sars-cov-2-spike-protein" class="btn btn--primary btn--large">Return to main text</a></p>

        
      </section>

      <footer class="page__meta">
        
        


        

      </footer>

      

      
    </div>

    
  </article>

  <!-- only show related on a post page when `related: true`
  
  
  -->
</div>

    </div>

    
      <div class="search-content">
        <div class="search-content__inner-wrap"><form class="search-content__form" onkeydown="return event.key != 'Enter';">
    <label class="sr-only" for="search">
      Enter your search term...
    </label>
    <input type="search" id="search" class="search-input" tabindex="-1" placeholder="Enter your search term..." />
  </form>
  <div id="results" class="results"></div></div>

      </div>
    

    <div id="footer" class="page__footer">
      <footer>
        <!-- start custom footer snippets -->

<!-- end custom footer snippets -->
        <div class="page__footer-follow">
  <hr>
  <a href="http://cbd.cmu.edu" target="_blank"><img align = "left" src="/biological_modeling_course/assets/images/combined_SCS_Unitmark_187_K.jpg" alt="CBD Logo" width="40%" height="40%"></a>
  <a href="http://mmbios.pitt.edu" target="_blank"><img align="right" src="/biological_modeling_course/assets/images/MMBioS_Logo.png" alt="MMBioS Logo" width="50%" height="50%"></a>
</div>

      </footer>
    </div>

    
  <script src="/assets/js/main.min.js"></script>




<script src="/assets/js/lunr/lunr.min.js"></script>
<script src="/assets/js/lunr/lunr-store.js"></script>
<script src="/assets/js/lunr/lunr-en.js"></script>







  </body>
</html>
